pkg/client/client.go

```diff
@@ -180,35 +179,31 @@ func Initialize(
 		tssNode,
 		operatorPublicKey,
 		keepsRegistry,
-		requestedSignatures,
+		eventDeduplicator,
 	)
 
 	// Watch for new keeps creation.
 	_ = ethereumChain.OnBondedECDSAKeepCreated(func(event *eth.BondedECDSAKeepCreatedEvent) {
 		logger.Infof(
-			"new keep [%s] created with members: [%x]\n",
+			"new keep [%s] created with members: [%x] at block [%d]",
 			event.KeepAddress.String(),
 			event.Members,
+			event.BlockNumber,
 		)
 
 		if event.IsMember(ethereumChain.Address()) {
 			go func(event *eth.BondedECDSAKeepCreatedEvent) {
-				if ok := requestedSigners.add(event.KeepAddress); !ok {
-					logger.Warningf(
-						"keep creation event for keep [%s] already registered",
+				if ok := eventDeduplicator.NotifyKeyGenStarted(event.KeepAddress); !ok {
```

Comment:
I prefer the `shouldHandle` name we used for the signing notification, it describes what's happening well.

```suggestion
				if shouldHandle := eventDeduplicator.NotifyKeyGenStarted(event.KeepAddress); !shouldHandle {
```

Reply:
Please see e194a9c36f2b28fc5b78968f16cc32fffeeed953.